tools/autograd/gen_trace_type.py

```diff
@@ -328,22 +323,24 @@ def emit_trace_body(f: NativeFunction) -> List[str]:
     dispatcher_sig = DispatcherSignature.from_schema(f.func)
     dispatcher_exprs = dispatcher_sig.exprs()
 
-    ret_and_arg_types = ', '.join([dispatcher_sig.returns_type()] + [a.type.cpp_type() for a in dispatcher_exprs])
     # code-generated tracing kernels plumb and recompute dispatch keys directly through the kernel for performance.
     # See Note [Plumbing Keys Through The Dispatcher] for details.
     dispatch_key_set = 'ks & c10::DispatchKeySet(c10::DispatchKeySet::FULL_AFTER, c10::DispatchKey::Tracer)'
-    redispatch_args = ', '.join(['op', dispatch_key_set] + [a.expr for a in dispatcher_exprs])
+    redispatch_args = ', '.join([dispatch_key_set] + [a.expr for a in dispatcher_exprs])
 
     assign_return_values = f'{tie_return_values(f)} = ' \
                            if f.func.kind() == SchemaKind.functional and f.func.returns else ''
 
+    api_name = cpp.name(f.func, faithful_name_for_out_overloads=True)
+    if f.manual_cpp_binding:
```

Comment:
`manual_cpp_binding` ops are hardcoded directly into `Functions.h`, so they aren't currently part of the redispatch API :( I opted instead to call their codegen'd `__dispatch_*`, which ARE available.

It would probably be nice if I could just call the fast-path variants instead. I opted not to for two reasons:
- I can't think of an elegant way to do it. One option is to additionally tell the codegen to generate variants in the redispatch API that directly call the hardcoded fast-path variant for ops that have `manual_cpp_binding` set. Another would just be to duplicate each hardcoded function from `Functions.h` into both namespaces.
- The perf benefit is probably minimal, since tracing kernels aren't called that frequently.

Reply:
Reasoning seems fine.

But consider constructing a full CppSignatureGroup and then extracting name from the faithful signature (if it exists). Helper method in CppSignatureGroup for returning faithful if it exists and regular if not will be helpful. Then you can axe the duplication of `__dispatch` logic.